operations/kubernetes/Troubleshoot_Intermittent_503s.md

```diff
@@ -1,33 +1,56 @@
 
    ```text
-   [2022-05-10T16:27:29.232Z] "POST /apis/hbtd/hmi/v1/heartbeat HTTP/2" 503 UF,URX "-" "TLS error: Secret is not supplied by SDS" 132 91 25 - "10.39.0.0" "-"
-   "08274ec4-2d93-4070-8fe1-fc6946f1cf26" "api-gw-service-nmn.local" "10.33.0.90:28500" outbound|80||cray-hbtd.services.svc.cluster.local - 10.33.0.43:443 10.39.0.0:29543 api-gw-service-nmn.local -
+   [2022-05-10T16:27:29.232Z] "POST /apis/hbtd/hmi/v1/heartbeat HTTP/2" 503 UF,URX "-" "TLS error: Secret is not supplied by SDS"
    ```
 
-1. Perform a rolling restart of the `istio-ingressgateway` pods:
+   Envoy container can occasionally get into this state when NCNs are being rebooted or upgraded, as well as when many deployments
+   are being created. When this happens, delete the pod if it is a single replica or perform a rolling restart of the Deployment
+   or StatefulSet if it is part of a multiple replica exhibiting the issue, like:
 
    ```bash
-   ncn-m # kubectl rollout restart -n istio-system deployment.apps/istio-ingressgateway
+   ncn-m# kubectl rollout restart -n istio-system deployment istio-ingressgateway
+   ```
+
+   Once the roll out is complete, the HTTP 503 message should clear.
+
+1. `UAEX` code:
+
+   The logs include errors similar to the following:
+
+   ```text
+   [2022-06-24T14:16:27.229Z] "POST /apis/hbtd/hmi/v1/heartbeat HTTP/2" 503 UAEX "-" 131 0 30 - "10.34.0.0" "-" "1797b0d3-56f0-4674-8cf2-a8a61f9adaea" "api-gw-service-nmn.local" "-" - - 10.40.0.29:443 10.34.0.0:15995 api-gw-service-nmn.local -
    ```
 
-1. Wait for the roll out to complete:
+   This error code typically indicates an issue with the authorization service (i.e., spire). When this happens, perform a rolling restart of the following:
 
    ```bash
-   ncn-m # kubectl rollout status -n istio-system deployment.apps/istio-ingressgateway
-   deployment "istio-ingressgateway" successfully rolled out
+   ncn-m# kubectl rollout restart -n spire statefulset spire-server
+   ncn-m# kubectl rollout restart -n spire daemonset spire-agent
+   ncn-m# kubectl rollout restart -n spire deployment spire-jwks
+   ncn-m# kubectl rollout restart -n spire deployment spire-postgres
+   ncn-m# kubectl rollout restart -n spire deployment spire-postgres-pooler
+   ncn-m# kubectl rollout restart -n spire daemonset request-ncn-join-token
    ```
 
-Once the roll out is complete, the HTTP 503 messages should clear, and the intermittent API behavior should resolve.
+   Once the roll out is complete, the HTTP 503 message should clear.
+
+1. Other error codes:
+
+   Although `UF,URX` and `UAEX` codes are most common, various other issues such as networking or application errors can cause different errors in
+   the pod or sidecar logs. Please refer to <https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#> for a list of
+   possible Envoy response flags. Generally you may try rolling restart of the application itself to see if it clears the error; otherwise, you need
```

Comment:
```suggestion
   possible Envoy response flags. In general, running a rolling restart of the application itself to see if it clears the error is a good practice; otherwise,
```